pandas/tseries/period.py

```diff
@@ -281,6 +281,16 @@ def __contains__(self, key):
             return False
         return key.ordinal in self._engine
 
+    def __array_wrap__(self, result, context=None):
```

Comment:
@sinhrks remember that ATM ``PeriodIndex`` is an odd duck. Its a real ``int64`` dtyped array, that happens to have boxing. So it allows any integer-like ops (as opposed to ``DatetimeIndex`` which prohibits a lot more because its ``M8[ns]``). 

So its needs handling to error on prohibited ops. So on on raising in ``__array_wraps__`` for prohibited ops, but this might require some dispatching mechanism to the sub-class to determine what is allowed. E.g. we do it similarly like this in the ``_add_numeric_ops`` and like routines.